pandas/core/strings.py

```diff
@@ -109,9 +118,77 @@ def cat_safe(list_of_columns: List, sep: str):
 
 def _na_map(f, arr, na_result=np.nan, dtype=object):
     # should really _check_ for NA
+    if is_extension_array_dtype(arr.dtype):
+        arr = extract_array(arr)
+        return _stringarray_map(f, arr, na_value=na_result, dtype=dtype)
     return _map(f, arr, na_mask=True, na_value=na_result, dtype=dtype)
 
 
+def _stringarray_map(
+    func: Callable[[str], Any], arr: "StringArray", na_value: Any, dtype: Dtype
+) -> ArrayLike:
+    """
+    Map a callable over valid elements of a StringArrray.
+
+    Parameters
+    ----------
+    func : Callable[[str], Any]
+        Apply to each valid element.
+    arr : StringArray
+    na_value : Any
+        The value to use for missing values. By default, this is
+        the original value (NA).
+    dtype : Dtype
```

Comment:
Elsewhere we say "np.dtype or ExtensionDtype".  Is this the new policy?

Reply:
Not sure if we have a policy. If it matters, this is an internal docstring, so I'm OK to use things from `pandas._typing` that we wouldn't have in a public docstring yet.